### PR TITLE
Refactor FXIOS-7650 [v121] Clean up content-blocker-lib-ios folder

### PR DESCRIPTION
**Pull request by lmarceau**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7650)

## :bulb: Description
- Remove `content-blocker-lib-ios` folder and move related code under Client
- Adjust scripts where needed
- Adjust gitignore and swiftlint
- This is what it now look like in Xcode:
Screenshot 2023-10-31 at 11 49 58 AM

Once this is merged, people might need to rerun `./bootstrap.sh` to get the content blocker files again. There's also the possibility that to make it work, derived data of the `ContentBlockerGenerator` needs to be deleted if an error `PCH was compiled with module cache path` happens.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
Some unit tests failed, I need to investigate

**Review by OrlaM — APPROVED** [review]

**Comment by lmarceau** [comment]
I'll wait for Isabel approval in case there's something else to change in here! 🙏

**Review by isabelrios — APPROVED** [review]

**Pull request merged**
